tools/dm/skip-replace-sqls.md

```diff
@@ -0,0 +1,610 @@
+            }
+        ]
+    }
+    ```
+
+    对应 DM-worker 节点中也可以看到类似如下日志：
+
+    ```bash
+    2018/12/28 15:33:13 operator.go:121: [info] [sql-operator] set a new operator 
+    uuid: c699a18a-8e75-47eb-8e7e-0e5abde2053c, pattern: ~(?i)ALTER\s+TABLE\s+`db2`.`tbl2`\s+DROP\s+COLUMN\s+`c2`, 
+    op: REPLACE, args: ALTER TABLE `db2`.`tbl2` DROP INDEX idx_c2; ALTER TABLE `db2`.`tbl2` DROP COLUMN `c2`
+    on replication unit
+    ```
+
+4. 在上游 MySQL 执行该 DDL 语句。
+
+5. 观察下游表结构是否变更成功，对应 DM-worker 节点中也可以看到类似如下日志：
+
+    ```bash
+    2018/12/28 15:33:45 operator.go:158: [info] [sql-operator] 
+    sql-pattern ~(?i)ALTER\s+TABLE\s+`db2`.`tbl2`\s+DROP\s+COLUMN\s+`c2` matched SQL 
+    USE `db2`; ALTER TABLE `db2`.`tbl2` DROP COLUMN `c2`;, 
+    applying operator uuid: c699a18a-8e75-47eb-8e7e-0e5abde2053c, 
+    pattern: ~(?i)ALTER\s+TABLE\s+`db2`.`tbl2`\s+DROP\s+COLUMN\s+`c2`, 
+    op: REPLACE, args: ALTER TABLE `db2`.`tbl2` DROP INDEX idx_c2; ALTER TABLE `db2`.`tbl2` DROP COLUMN `c2`
+    ```
+
+6. 使用 `query-status` 确认该任务的 `stage` 持续为 `Running`。
+
+7. 使用 `query-error` 确认已不存在 DDL 执行错误。
+
+#### 合库合表场景下同步中断后被动执行跳过操作
+
+##### 应用场景
+
+假设现在通过多个 DM-worker 将上游多个 MySQL 实例内的多个表进行合库合表同步到下游 TiDB 的同一个表，并且上游各分表执行了 TiDB 不支持的 DDL 语句。
+
+DM-master 通过 DDL lock 协调 DDL 同步、并请求 DDL lock owner 向下游 TiDB 执行该 DDL 语句后，由于 TiDB 不支持该 DDL 语句，同步任务会报错并中断。
+
+##### 被动跳过 SQL 语句
+
+合库合表场景下，被动跳过 TiDB 不支持的 DDL 语句的处理方式与非合库合表场景下的[同步中断后被动执行跳过操作](#同步中断后被动执行跳过操作)基本一致。
+
+但在合库合表场景下，只需要 DDL lock owner 向下游同步该 DDL 语句，因此在两种场景下的处理过程主要存在以下区别：
+
+1. 合库合表场景下，仅需要对 DDL lock owner 执行 `sql-skip`（`--worker={DDL-lock-owner}`）。
+
+2. 合库合表场景下，仅需要对 DDL lock owner 执行 `resume-task`（`--worker={DDL-lock-owner}`）。
+
+#### 合库合表场景下同步中断前主动执行替代执行操作
+
+##### 应用场景
+
+假设现在存在如下四个上游表需要合并同步到下游的同一个表 ``` `shard_db`.`shard_table` ```：
+
+- MySQL 实例 1 内有 `shard_db_1` 逻辑库，包括 `shard_table_1` 和 `shard_table_2` 两个表。
+- MySQL 实例 2 内有 `shard_db_2` 逻辑库，包括 `shard_table_1` 和 `shard_table_2` 两个表。
+
+初始时表结构为：
+
+```sql
+mysql> SHOW CREATE TABLE shard_db_1.shard_table_1;
++---------------+------------------------------------------+
+| Table         | Create Table                             |
++---------------+------------------------------------------+
+| shard_table_1 | CREATE TABLE `shard_table_1` (
+  `c1` int(11) NOT NULL,
+  `c2` int(11) DEFAULT NULL,
+  PRIMARY KEY (`c1`),
+  KEY `idx_c2` (`c2`)
+) ENGINE=InnoDB DEFAULT CHARSET=latin1 |
++---------------+------------------------------------------+
+```
+
+此时，在上游所有分表上都执行以下 DDL 操作修改表结构（即 DROP 列 c2）：
+
+```sql
+ALTER TABLE shard_db_*.shard_table_* DROP COLUMN c2;
+```
+
+则当 DM 通过 sharding DDL lock 协调两个 DM-worker 同步该 DDL 语句、并请求 DDL lock owner 向下游执行该 DDL 语句时，会由于 TiDB 不支持该 DDL 语句而导致同步任务中断且报如下错误：
+
+```bash
+exec sqls[[USE `shard_db`; ALTER TABLE `shard_db`.`shard_table` DROP COLUMN `c2`;]] failed,
+err:Error 1105: can't drop column c2 with index covered now
+```
+
+**但如果在上游实际执行该 DDL 语句前，你已提前知道该 DDL 语句不被 TiDB 所支持。**则可以使用 `sql-skip` 或 `sql-replace` 命令为该 DDL 语句预设一个跳过/替代执行操作。
```

Comment:
```suggestion
**但如果在上游实际执行该 DDL 语句前，你已提前知道该 DDL 语句不被 TiDB 所支持。** 则可以使用 `sql-skip` 或 `sql-replace` 命令为该 DDL 语句预设一个跳过/替代执行操作。
```

Reply:
@yikeke need to address this comment to make `**` work.

Reply:
Updated. PTAL again. @csuzhangxc @lilin90